clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -2813,6 +2813,11 @@ void CodeGenFunction::EmitKCFIOperandBundle(
       Callee.getAbstractInfo().getCalleeFunctionProtoType();
   if (FP)
     Bundles.emplace_back("kcfi", CGM.CreateKCFITypeId(FP->desugar()));
+  else {
+    ASTContext &context = this->getContext();
+    QualType voidType = context.VoidTy;
```

Comment:
`VoidTy` doesn't seem like an appropriate type for a function even if it doesn't have a parameter type list. We still want to distinguish between different return types, right?

Also, please add a matching test when you make a change. You could probably add a test case for this to `clang/test/CodeGen/kcfi.c`.